File: E-Commerce App Project (Tabbed)/Controllers/ThankYouViewController.m
//
//  ThankYouViewController.m
//  E-Commerce App Project (Tabbed)
//
//  Created by Rony Banik on 18/11/18.
//  Copyright © 2018 Rony Banik. All rights reserved.
//

#import "ThankYouViewController.h"

@interface ThankYouViewController ()

@end

@implementation ThankYouViewController
@synthesize wkwebviewOutletForPDFShow,uiviewForArko,emailThankYouPage;

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    NSString *PDFFilePath = [defaults valueForKey:@"pdfFilePath"];
    emailThankYouPage.text = [defaults valueForKey:@"SessionLoggedInuserEmail"];
    
    NSURL *pdfUrl = [[NSURL alloc]initFileURLWithPath:PDFFilePath];
    
    [wkwebviewOutletForPDFShow loadFileURL:pdfUrl allowingReadAccessToURL:pdfUrl];
    
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: E-Commerce App Project (Tabbed)/Controllers/ThankYouViewController.swift
//
//  ThankYouViewController.swift
//  E-Commerce App Project (Tabbed)
//
//  Converted to Swift with MVVM pattern
//

import UIKit
import WebKit

class ThankYouViewController: UIViewController {
    
    // MARK: - ViewModel
    private let viewModel = ThankYouViewModel()
    
    // MARK: - IBOutlets
    @IBOutlet var wkwebviewOutletForPDFShow: WKWebView!
    @IBOutlet var uiviewForArko: UIView!
    @IBOutlet var emailThankYouPage: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        bindViewModel()
    }
    
    private func bindViewModel() {
        viewModel.userEmail.bind { [weak self] email in
            self?.emailThankYouPage.text = email
        }
        
        viewModel.pdfFilePath.bind { [weak self] path in
            guard let self = self, let pdfUrl = self.viewModel.getPDFURL() else { return }
            self.wkwebviewOutletForPDFShow.loadFileURL(pdfUrl, allowingReadAccessTo: pdfUrl)
        }
    }
}
